### PR TITLE
Add Film-Detail endpoint to the API spec

### PR DESCRIPTION
**Pull request by gstro**
This PR adds the specification for details for a particular film identified by its ID. 
- Specifies a `GET` to `/films/{id}` 
- Specifies the `FilmDetail` payload and its subtypes
- Includes other cleanup like improved names and more specific requirements
- Removes the 'swaggerhub' details since trial expired 😅, and renames file

Because of the rename, the diff isn't that helpful. You can view a better diff by comparing the last api-spec commit to the tip of main: https://github.com/wds9601/film-club/compare/3c02206...c21de12

Will follow up with README update to link to new file.

### CODE REVIEW

**Review by wds9601 — APPROVED** [review]
rad. thanks.

File: README.md [hunk 1]
```diff
@@ -32,7 +32,7 @@ These were the new tools I used developing FilmClub:
 ## Getting Started
 
 1. Clone and follow directions to start the corresponding **FilmClub** [backend server](https://github.com/gstro/film-club-server)   
-*(currently private, refer to our [OpenAPI specs](https://github.com/wds9601/film-club/blob/main/yaml-unresolved/swagger.yaml) for API schema)*
+*(currently private, refer to our [OpenAPI specs](https://github.com/wds9601/film-club/blob/main/specs/filmclub.yaml) for API schema)*
 3. Clone this repo to local machine
 2. Navigate to `film-club-client` folder
 4. In project folder:
```
Comment (gstro) [inline README.md:38]: There's more in the README I should change 👇, about the `yaml unresolved` folder.

File: specs/filmclub.yaml [hunk 2]
```diff
@@ -0,0 +1,793 @@
...
+            minimum: 1
+      responses:
+        "200":
+          description: List of upcoming films with pagination info
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/UpcomingFilmsResponse'
+  
+  /films/images/{imageType}/{imagePath}:
```
Comment (gstro) [inline specs/filmclub.yaml:148]: Now that we have cast and crew profile image paths, I'm not sure this is good design for this endpoint 🤔

Comment (gstro) [reply specs/filmclub.yaml:148]: Ah, found [this from the TMDB API](https://developers.themoviedb.org/3/configuration/get-api-configuration), I think we can just add more `imageType`s (i.e. `poster`, `backdrop`, `logo`, `profile`, `still`).
```
  {
    "backdrop_sizes": [
      "w300",
      "w780",
      "w1280",
      "original"
    ],
    "logo_sizes": [
      "w45",
      "w92",
      "w154",
      "w185",
      "w300",
      "w500",
      "original"
    ],
    "poster_sizes": [
      "w92",
      "w154",
      "w185",
      "w342",
      "w500",
      "w780",
      "original"
    ],
    "profile_sizes": [
      "w45",
      "w185",
      "h632",
      "original"
    ],
    "still_sizes": [
      "w92",
      "w185",
      "w300",
      "original"
    ]
  }
```

File: specs/filmclub.yaml [hunk 3]
```diff
@@ -0,0 +1,793 @@
...
+        - releaseDate
+        - status
+        - title
+        - voteAverage
+        - voteCount
+        - images
+        - videos
+        - credits
+        - similar
+        - watchProviders
```
Comment (gstro) [inline specs/filmclub.yaml]: Not `required`! 😬

File: specs/filmclub.yaml [hunk 4]
```diff
@@ -0,0 +1,793 @@
...
+        name:
+          type: string
+        site:
+          type: string
+        size:
+          type: integer
+          enum: [360, 480, 720, 1080]
+        type:
+          type: string
+          enum: [Trailer, Teaser, Clip, Featurette, Behind the Scenes, Bloopers]
```
Comment (gstro) [inline specs/filmclub.yaml:686]: Could be good for grouping on the Film Detail page.

File: specs/filmclub.yaml [hunk 5]
```diff
@@ -0,0 +1,793 @@
...
+        {
+          "id": "533ec654c3a36854480003eb",
+          "iso_639_1": "en",
+          "iso_3166_1": "US",
+          "key": "SUXWAEX2jlg",
+          "name": "Trailer 1",
+          "site": "YouTube",
+          "size": 720,
+          "type": "Trailer",
+          "url": "https://www.youtube.com/watch?v=SUXWAEX2jlg"
```
Comment (gstro) [inline specs/filmclub.yaml:700]: Should change to the _embedded_ url. I'll look for **YouTube** and **Vimeo**...

Comment (gstro) [reply specs/filmclub.yaml:700]: Ok, **YouTube** would be `https://www.youtube.com/embed/{key}`

Comment (gstro) [reply specs/filmclub.yaml:700]: Looks like **Vimeo** would be `https://player.vimeo.com/video/{key}`

File: specs/filmclub.yaml [hunk 6]
```diff
@@ -0,0 +1,793 @@
...
+servers:
+  - description: local
+    url: 'http://localhost:8080/v1'
+tags:
+  - name: echo
+    description: Useful for healthchecks and testing connections.
+  - name: film
+    description: Info about a film
+paths:
+  /echo/hello/{name}:
```
Comment (gstro) [inline specs/filmclub.yaml:18]: Considering getting rid of these testing endpoints... Maybe at least a health-check?

File: specs/filmclub.yaml (hunk 2 above)
Comment (gstro) [reply specs/filmclub.yaml:148]: Can follow up in separate PR.

File: specs/filmclub.yaml (hunk 5 above)
Comment (gstro) [reply specs/filmclub.yaml:700]: I'll keep the `url` key in the spec and just change server implementation to construct the embedded url for the value.

**Review by gstro — COMMENTED** [review]
Some notes for me to follow up.

File: specs/filmclub.yaml (hunk 2 above)
Comment (gstro) [reply specs/filmclub.yaml:148]: #34

**Pull request merged**